Components/Queues/PhpAmqpLibQueueManager.php

```diff
@@ -0,0 +1,309 @@
+                shuffle($this->connections);
+            }
+
+            $this->connection = null;
+            $tested = [];
+
+            foreach ($this->connections as $connection) {
+                try {
+                    $connection->connect();
+                    $this->connection = $connection;
+                    break;
+                } catch (\AMQPConnectionException $e) {
+                    $tested[] = "{$connection->getHost()}: {$e->getMessage()}";
+                }
+            }
+
+            if (!$this->connection) {
+                throw new \RuntimeException(sprintf('Unable to connect to any of the following hosts:%s%s', PHP_EOL, implode(PHP_EOL, $tested)));
+            }
+        }
+
+        if (!$this->channel) {
+            $this->declareChannel();
+        }
+    }
+
+    /**
+     * Destroys the connection with the queuing system.
+     */
+    public function disconnect()
+    {
+        try {
+            if ($this->isConnected()) {
+                $this->connections[0]->close();
+            }
+            $this->connection = null;
+        } catch (\Exception $exception) {
+            throw new \Exception($exception->getMessage());
+        }
+    }
+
+    /**
+     * Returns true if a connection already exists with the queueing system, false otherwise.
+     * @return bool
+     */
+    public function isConnected(): bool
+    {
+        return $this->connections[0]  ? $this->connections[0]->isConnected() : false;
+    }
+
+    /**
+     * Declares the queue to drive the message
+     *
+     * @param string $name      The name of the que
+     * @param int    $flag      See AMQPQueue::setFlags()
+     * @param array  $arguments See AMQPQueue::setArguments()
+     *
+     * @return \AMQPQueue
+     *
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function declareQueue(string $name, int $flag = AMQP_DURABLE, array $arguments = [])
+    {
+        if (!$this->channel) {
+            throw new \AMQPChannelException('There is no channel available');
+        }
+
+        $this->queue = $name;
+        $durable = $flag === AMQP_DURABLE ? true : false;
+        return $this->channel->queue_declare($this->queue, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Publish a message
+     * @param AMQPMessage $message
+     */
+    public function publish(AMQPMessage $message)
+    {
+        $this->channel->basic_publish($message, self::EXCHANGE_NAME, $this->queue);
+    }
+
+    /**
+     * Add a connection to the browker
+     * @param $connection
+     * @return $this
+     */
+    public function addConnection($connection): self
+    {
+        $this->connections[] = $connection;
+
+        return $this;
+    }
+
+    /**
+     * Returns all the connections available
+     * @return \AMQPConnection[]|array|null
+     */
+    public function getConnections():? array
```

Comment:
Where this method is used? I don't see any place where is used in the code

Reply:
This class was removed.